### PR TITLE
JS

### PR DESCRIPTION
**Pull request by majac91**
### Learnings
**.map** vs **.forEach** 

- Just about anything you can do with forEach() you can do with map(), and vise versa.
- map() allocates memory and stores return values. forEach() throws away return values and always returns undefined.
-  forEach() will allow a callback function to mutate the current array. map() will instead return a new array.

**Imperative vs. Declarative JavaScript** 

**.target**  vs. **.currentTarget**

.target = element that triggered event.
.currentTarget = element that has the event listener.

**Don't skip heading levels (`<h1>-<h6>`)**

**Marking activatable labels using aria-pressed**
Buttons that serve as a "toggle" (have an active and not active state) should have an aria-pressed="true" attribute (true or false)

### References
**Debugging JavaScript** https://www.youtube.com/watch?v=H0XScE08hy8&ab_channel=GoogleChromeDevelopers
**.map** vs **.forEach**  vhttps://codeburst.io/javascript-map-vs-foreach-f38111822c0f
**Imperative vs. Declarative JavaScript**  https://dzone.com/articles/imperative-vs-declarative-javascript#:~:text=Imperative%20and%20Declarative%20Styles,is%20that%20you%20want%20done.
**Headings and landmarks** https://web.dev/headings-and-landmarks/
**Aria-pressed** https://www.accessibility-developer-guide.com/examples/sensible-aria-usage/pressed/

### CODE REVIEW

**Comment by majac91** [comment]
I'm stuck trying to attach the correct url to each element, the following doesn't seem to work 
```
      for(let i=0; i<destinations.length; i++) {
        img.src = url[i];
      };
```

Also I'm still getting ```ReferenceError: document is not defined``` when running code in editor.

**Comment by majac91** [comment]
I want to assign display: none to  and the sibiling <figcaption> but I'm getting something wrong trying to select the caption, here's what I've tired: 
```buttonVisited.addEventListener('click', () => {
  const visited = document.querySelectorAll('.visited');
  // const captions = document.querySelectorAll('.visited .caption');   
     const captions = visited.forEach(item => {
     return item.nextElementSibling;
  });
.........
});
```

File: src/main.js [hunk 1]
```diff
@@ -244,3 +244,81 @@ let destinations = [
...
+  const getURL = (destinations) => {
+    let url= destinations.map(obj => obj.photo.valueOf());
+    return url;
+  }
+
+  //create elements for each city
+  const elements= destinations.forEach(destination => {
+
+    //picture element
+    const createImg = (url) => {
```
Comment (sandrina-p) [inline src/main.js]: One tip, about VSCode. It seems like you don't have the default VSCode Eslint/Prettier enabled. It's __very__ useful to make your code pretty as you write it (correct indentation, consistency in spacings, quotes, etc...)

Install [Prettier](https://prettier.io/) and [Eslint](https://eslint.org/) VSCode extension
image
image

Go to VSCode settings, tick "Format of Save" and the restart VSCode.

image

File: src/main.js [hunk 2]
```diff
@@ -244,3 +244,81 @@ let destinations = [
...
+      let figure = document.createElement('figure');
+    //img element
+      let img = document.createElement('img');
+      img.style.objectFit ='cover';
+      document.body.appendChild(figure);
+      figure.appendChild(img);
+    //img caption
+      let figcaption= document.createElement('figcaption');
+      figure.appendChild(figcaption);
+      figcaption.innerText= `${destinations.city}, ${destinations.country}`
```
Comment (sandrina-p) [inline src/main.js]: Here `destinations.city` does not work. It should be `destination.city`. The first is a list (array) the other is the destination itself (the object).

A tip to avoid this mistakes. Name lists like this `destinationList` `animalList` etc... that what you detect immediately if it's a list or not ;)

File: src/main.js [hunk 3]
```diff
@@ -244,3 +244,81 @@ let destinations = [
...
+
+
+      //get urls
+      for(let i=0; i<destinations.length; i++) {
+        img.src = url[i];
+      };
```
Comment (sandrina-p) [inline src/main.js]: This loop is not needed (and wrong). You can do directly `img.src = destination.photo`. What made u think you needed to do a loop here?

File: src/main.js [hunk 4]
```diff
@@ -244,3 +244,81 @@ let destinations = [
...
+  const buttonVisited = document.createElement('button');
+  buttonVisited.innerText='Visited';
+  buttonVisited.type='button';
+  document.body.appendChild(buttonVisited);
+  
+  
+  const buttonBucketlist = document.createElement('button');
+  buttonBucketlist.innerText='Bucketlist';
+  buttonBucketlist.type='button';
+  document.body.appendChild(buttonBucketlist);
```
Comment (sandrina-p) [inline src/main.js]: These buttons are not dynamic like the images, they could be done directly on HTML side ;)

**Review by sandrina-p — COMMENTED** [review]
This is looking goood!

File: src/main.js [hunk 5]
```diff
@@ -244,3 +244,81 @@ let destinations = [
...
+
+  return createImg(getURL(destinations));
+
+});
+
+  //create buttons
+  const buttonVisited = document.createElement('button');
+  buttonVisited.innerText='Visited';
+  buttonVisited.type='button';
+  document.body.appendChild(buttonVisited);
```
Comment (sandrina-p) [inline src/main.js]: > Also I'm still getting ReferenceError: document is not defined when running code in editor.

Where exactly do you get that error? Can you show a screenshot and the exact steps? You are running the project with `npm run dev`, right?

File: src/main.js [hunk 6]
```diff
@@ -244,3 +244,81 @@ let destinations = [
...
+  const buttonBucketlist = document.createElement('button');
+  buttonBucketlist.innerText='Bucketlist';
+  buttonBucketlist.type='button';
+  document.body.appendChild(buttonBucketlist);
+  
+
+//connect lists with buttons
+
+buttonVisited.addEventListener('click', () => {
+  document.querySelectorAll('.visited');
```
Comment (sandrina-p) [inline src/main.js]: To do this, instead of your approach (that is complex), it's easier if we add a class to the list parent and then with CSS you manipulate the list items based on that class. For example:

```html
<div class="list">
  <figure class="visited">...</figure>
  <figure class="bucketList">...</figure>
  <figure class="visited">...</figure>
</div>
```
// with JS add a class ".visited" to list when clicking the button. so The DOM is like this:
```html
<div class="list visited">
  ....
</div>
```

... and with CSS you hide the elements based on the list class:
```css
.list.visited figure {
  display: none; // Hide all items (figure) when .list has .visited class
} 

.list.active figure.visited {
  display: block; // overide the figure.visited items to be visible
}
```

This is where CSS selectors real powers come into action! Good luck!

File: src/main.js [hunk 7]
```diff
@@ -244,3 +244,81 @@ let destinations = [
   ]
 
 
+  //visited and bucktelist lists
```
Comment (sandrina-p) [inline src/main.js]: With Parcel (and many any other modern module bundler tools) the way images are load is quite different. They need to be "required" so that the local server (localhost) know exactly where they are.

For that we use `require()` and pass the image path. Do not forget the `./` at the beginning so that Parcel knows the path is related to the `main.js`:

```diff 
  {
    city: "Rome",
    country: "Italy",
+    photo: require("./img/rome.jpg"),
    visited: true,
  },
```

File: src/main.js (hunk 5 above)
Comment (majac91) [reply src/main.js]: It's just in the editor when I try to use an extension to run js (with npm running), but in the browser it seems to work fine
Screenshot 2020-09-05 at 16 55 11

File: src/main.js (hunk 3 above)
Comment (majac91) [reply src/main.js]: The idea was: I have the variable ```url```(a list of all urls) so I wanted to loop through ```destinations``` and for each object in an array assign the ```img.scr```  the correct link, it should get the index of each element that goes through the loop. I get that it's wrong, just not why?

File: src/main.js (hunk 5 above)
Comment (sandrina-p) [reply src/main.js]: "I try to use an extension to run js (with npm running)". What extension? What do you use it for?

File: src/main.js (hunk 3 above)
Comment (sandrina-p) [reply src/main.js]: `url[i]`, does indeed get the link correctly. (If you console.log it inside the loop, you'll see it)
But you are assigning each url to the same thing `img.src`, overriding it over and over again with different URLs. In the end, the result is the URL of the last index of the loop.

Comment (majac91) [reply src/main.js]: Ohhh now I see. So... I need a new instance of img.src for each city? It's something I remember in relation to classes, does it make sense in this context? Or another loop?

File: src/main.js (hunk 5 above)
Comment (majac91) [reply src/main.js]: This one, I just used it to run js in vscode instead of browser console 
Screenshot 2020-09-05 at 17 42 36

File: src/main.js (hunk 6 above)
Comment (majac91) [reply src/main.js]: It does sound much less complicated, I'll redo  it. This is the code I forgot to push, I'm curious why I wasn't able to select the .visited sibiling (two ways I tried are on lines 3-5) 

```
 buttonVisited.addEventListener('click', () => {
  const visited = document.querySelectorAll('.visited');
  // const captions = document.querySelectorAll('.visited .caption');   
  const captions = visited.forEach(item => {
    return item.nextElementSibling;
  })

  //add display:none to each img and figcaption 
  visited.forEach(item => {
    item.style.display='none';
  })
  captions.forEach(item => {
    item.style.display='none';
  });

});
```

File: src/main.js (hunk 5 above)
Comment (sandrina-p) [reply src/main.js]: Oh, but this is to run server-side code, and there document does not exist (it's something that only exists in browsers).
Anyway, here, you should run the code in developer tools and learn how to use debugger and breakpoints there. It will help u a lot debugging the code ;) [Here's a chrome tutorial](https://www.youtube.com/watch?v=H0XScE08hy8&ab_channel=GoogleChromeDevelopers) (it also works for firefox!)

File: src/main.js (hunk 3 above)
Comment (sandrina-p) [reply src/main.js]: Maybe you are still confused (or I didn't understand your goal).
In this scope, we just need to worry about one image per destination, right?. We are already within the loop (line 260) and in this specific destination details. So we do not need to do a new loop. Does it make sense?

Comment (majac91) [reply src/main.js]: Wow I really overcomplicated there 😄  ```img.src = destination.photo``` worked like a charm 🙌🏻

File: src/main.js (hunk 6 above)
Comment (sandrina-p) [reply src/main.js]: This is because `forEach` is being used incorrectly. I recommend you re-reading again the [diff between `forEach()` and `map()`](https://codeburst.io/javascript-map-vs-foreach-f38111822c0f). In this case, you can use both. Personally I'd use map because I like [declarative paradigm](https://dzone.com/articles/imperative-vs-declarative-javascript#:~:text=Imperative%20and%20Declarative%20Styles,is%20that%20you%20want%20done.).

I challenge you to try both approaches (forEach and map) to get the `.captions`.Try it, before I share the solution with you ;)

Comment (majac91) [reply src/main.js]: Just saw this now after moving away from that solution but trying map with my current one didn't work (because it returns a new array and I need the unaltered one..?)

I wasn't really getting this part 
```
// with JS add a class ".visited" to list when clicking the button. so The DOM is like this:

<div class="list active">
  ....
</div>
```
so I tried figuring it out on my own and it worked! What I don't like about it is that I had to copy the same variables in each block, is there a better way to access them? 
Anyways I'd love to know how you do this so if you could expand on the example above that would be awesome

**Comment by majac91** [comment]
Any idea why I have this happening for these three items? @sandrina-p 

Screenshot 2020-09-06 at 17 58 13

File: src/fonts/@font-face/Demo.html [hunk 8]
```diff
@@ -0,0 +1,95 @@
+<!DOCTYPE html>
```
Comment (sandrina-p) [inline src/fonts/@font-face/Demo.html]: I believe you don't need this file and other files in this folder. Just add the font file itself and that's all good :)

File: src/main.js [hunk 9]
```diff
@@ -177,70 +177,255 @@ let destinations = [
...
+ 
+
+  //get list of urls
+  const getURL = (destinationsList) => {
+    let url= destinationsList.map(obj => obj.photo.valueOf());
+    return url;
+  }
+
+  //create elements for each city
+  const elements= destinationsList.forEach(destination => {
```
Comment (sandrina-p) [inline src/main.js]: This `elements` has nothing. A `forEach` does not return anything.

File: src/main.js (hunk 6 above)
Comment (sandrina-p) [reply src/main.js]: > (because it returns a new array and I need the unaltered one..?)

It returns a new array but you still have access to the original one (it isn't mutated)


> What I don't like about it is that I had to copy the same variables in each block, is there a better way to access them?


I think you didn't understand it fully, I'm sorry, I mean "list visited", not "list active" :(
You just need to add `. visited ` or `.bucketlist` to the items list parent (eg `.gallery visited`), not to each list item. That way the code will be much smaller.

And then, with CSS you do the display block/none as I showed in the snippet above.

Finally, even with smaller code, you can still create an extra function `addClassToGallery()` that accepts a class name. That way you avoid duplicating the variables ;)

I'll share a solution soon, don't worry. Just give it a try again based on my new indications. ;)

Comment (majac91) [reply src/main.js]: Still confused about that snippet, why ```.active``` here?
```
.list.active figure.visited {
  display: block; // overide the figure.visited items to be visible
}
```

Comment (sandrina-p) [reply src/main.js]: It was not `.active`, it was `.visited`. I did a small demo showing this filter with css technique. I hope it helps you to understand how to apply it in this project ;)

https://codepen.io/sandrina-p/pen/OJNQJbz

File: src/main.js (hunk 9 above)
Comment (majac91) [reply src/main.js]: Not sure what to do about it since deleting it breaks the gallery?

Comment (sandrina-p) [reply src/main.js]: You already did it :D Now there's just the `destinationsList.forEach()` without assigning it to a const, and that's the right way of doing it ;)

File: src/main.js (hunk 1 above)
Comment (sandrina-p) [reply src/main.js]: @majac91 I wonder if you were able to do this? Last time I checked the code it's still not properly indented 🤔

File: src/main.js [hunk 10]
```diff
@@ -298,108 +298,38 @@ let destinationsList = [
...
-//                 clicked.currentTarget.classList.remove('active-button');
-//     };
-// }
-// buttonRemoveActive();
+  //get clicked button
+  const buttonIsActive = () => {
+    const buttons= document.querySelectorAll('.button');
+    buttons.forEach(button => {
+        button.addEventListener('click', clicked => {
+            clicked.currentTarget.classList.add('active-button');
```
Comment (sandrina-p) [inline src/main.js]: When another btn is clicked, it should remove this class from the other buttons. Otherwise all bts will have `.active-button`.

Comment (majac91) [reply src/main.js]: I was just about to ask you for a hint about that :D is there a way to get the previous button and remove .active from it?

File: src/css/style.scss [hunk 11]
```diff
@@ -1,3 +1,170 @@
...
+    font-family: 'GermanoRegular';
+    font-weight: normal;
+    font-style: normal;
+}
+
+//shrink navbar on scroll
+.shrink {
+    position: fixed;
+    font-size: 1em;    
+    height: 10vh;
```
Comment (sandrina-p) [inline src/css/style.scss]: You need to ensure this scroll nav transition on scroll is smooth. By replacing between `fixed` and `relative` the code will jump around each time it changes. My suggestion: Start with `fixed` by default and adjust the rest of the content based on it ;)

tip 2: Why using height 10vh? The proportions might not be as expected. Usually here you want something more constant, such as `5rem`

File: src/main.js (hunk 10 above)
Comment (majac91) [reply src/main.js]: I tried toggle and add/remove but didn't get it right

File: src/index.html [hunk 12]
```diff
@@ -3,14 +3,27 @@
...
+        <nav class='navbar '>
+            <h4 class="navbar__title">places been</h4>
+            <h4 class='navbar__subtitle'>a travel journal</h4>
+        </nav>
+        <div class="buttons container">
+            <h4 class=buttons__view>View:</h4>
+            <button class='button visited-button' type='button'>Visited</button>
+            <button class='button bucketlist-button' type='button'>Bucketlist</button>
+            <button class='button all' type='button'>Show all</button>
+        </div>
```
Comment (sandrina-p) [inline src/index.html]: A little about A11Y (Accessibility):

Headings should follow a semantic structure, starting from h1 up to h6 and should be used in a meaningful way. Don't use a heading just because it "looks like" a heading. My general tip to find the best heading for content is to imagine the page without styles, as if it was a book. What heading would you use? This is something that comes with experience though.

Here's a short [google article about headings and landmarkst](https://web.dev/headings-and-landmarks/).

When you are done reviewing the structure of this page, let me know so that I can give further feedback ;)

File: src/main.js (hunk 10 above)
Comment (sandrina-p) [reply src/main.js]: You'll need to loop through the buttons inside the click eventlistener. Loop through them and remove the class. Then target the current one and add the list.

Tip: It's a common thing to call event listener parameter as `event`, because we receive info about that event. calling it `clicked` might be confusing/misleading.

File: src/css/style.scss (hunk 11 above)
Comment (majac91) [reply src/css/style.scss]: Yeah I have no idea what the difference is between e.g. using vh vs. rem in this case, I kinda used it arbitrarily, I'll read on that

> Start with fixed by default and adjust the rest of the content based on it ;)

adjust in what way? What exactly needs fixing, the sudden transition when you start scrolling or..?

Comment (sandrina-p) [reply src/css/style.scss]: `vh` changes based on the browser height. So fix the browser is small, the header height will be smaller.
`rem` adjusts based on the user zoom settings. The normal is 16px, so that does not change that much.

> adjust in what way? What exactly needs fixing, the sudden transition when you start scrolling or..?

Let me put it in another way: Don't animate the position. Start with `fixed` by default and let it be even when scrolled ;)

File: src/main.js (hunk 9 above)
Comment (majac91) [reply src/main.js]: Oh I see 😄 I didn't get you correctly

File: src/main.js (hunk 10 above)
Comment (majac91) [reply src/main.js]: Noted on the parameter 

>  Loop through them and remove the class. Then target the current one and add the list.

I'm not getting the order of this. You mean remove the active class?

Comment (sandrina-p) [reply src/main.js]: Here's a pseudo-code to help you:

```
// add click event to each button
//      loop through each button
//              remove the active class
//      add active class to the current button
``

Comment (majac91) [reply src/main.js]: But why remove it and _then_ add? Why loop once to add click to each button and then again through each button?

Comment (majac91) [reply src/main.js]: I might need to get back to this one later, I feel like I'm missing something fundamental here

Comment (majac91) [reply src/main.js:328]: This is the closest I got, with some unexpected results 😄 

`buttons.forEach(button => {
  button.addEventListener('click', event => {
    current = event.currentTarget;
    for(let i=0; i<buttons.length && button!==current; i++) {

        button.classList.remove('active-button');
    }
    button = event.currentTarget.classList.add('active-button');
  });
});`

Comment (majac91) [reply src/main.js]: So a break helped, I think I got it right 😄 Not a 100% sure why exactly it works, but it works

Comment (sandrina-p) [reply src/main.js]: Yey, you made it worked indeed! One cool thing about JS is that there are a lot of different ways to approach something! It wasn't exactly as I suggested but the goal is completed and that's what matters now.


After a break, I also had a better idea to approach this without needing to use a loop (which indeed is too much overload). The idea is to store a variable `elNavBtnActive` that contains the current clicked button. Then, in the btn click event, use that variable to remove the active class and replace the variable with the recent clicked nav button. That way we don't need to loop through the buttons to remove the active class.

Try to come up with a solution (really try). If after 10min you didn't figure it out, here's the solution down below:

<details>
<summary>Try by yourself before seeing the solution here</summary>

<details>
<summary>I hope you really did try! Here it goes..</summary>

```js
const buttons = document.querySelectorAll(".button");
let elButtonActive = null; // initially there isn't any active/clicked button, so this is null

buttons.forEach((button) => {
  button.addEventListener("click", (event) => {
    const clickedButton = event.target;
    
    // If there's a current actibe button, remove the class
    if (elButtonActive) elButtonActive.classList.remove("active-button");

    // Add active class to the current clicked button
    clickedButton.classList.add("active-button");

    // Replace the variable with the most recent (current) clicked button
    elButtonActive = clickedButton;
  });
});
```


</details>

</details>

File: src/index.html (hunk 12 above)
Comment (majac91) [reply src/index.html]: Done. I had doubts whether to leave ```nav``` or replace it with ```header``` but I suppose it will have a navigation functionality at one point

File: src/main.js (hunk 10 above)
Comment (majac91) [reply src/main.js]: Here's what didn't work, at least it was in somewhat of a good direction. It's great having something to compare with as in most cases when I get something wrong it's hard to tell why it's that way and how to fix it. Your example here gives me something to learn from 

```
buttons.forEach(button => {
  button.addEventListener('click', event => {
  const elNavBtnActive = event.currentTarget;

  if(!elNavBtnActive) {
    elNavBtnActive=elNavBtnActive.className.replace(" active-button", "");
  }
  elNavBtnActive.classList.add('active-button')
});
});
```

File: src/index.html (hunk 12 above)
Comment (majac91) [reply src/index.html]: Any additional feedback here?

File: src/main.js (hunk 10 above)
Comment (sandrina-p) [reply src/main.js]: So you got it almost, the problem was the scope of `elNavBtnActive`. It was created inside the event listener, so each time the click happens, a new variable is created, forgetting the value stored in the previous click. That's why `elNavBtnActive` must be created in an outside scope (as in my example).

Does it make sense? I recommend you to review the concept of scoping in JS :)

File: src/index.html (hunk 12 above)
Comment (sandrina-p) [reply src/index.html]: The `nav` should indeed be a `header`. It's not a nav because it does not contain anything related to navigation ATM. In the future, we can create a `nav` inside the `header`.

File: src/main.js [hunk 13]
```diff
@@ -5,170 +5,170 @@ var dateFormating = moment(myDate).format('LL');
```
Comment (sandrina-p) [inline src/main.js]: @majac91 were you able to setup ESLint + Prettier so that VSCode automatically formats the code on save? Please do it before merging this PR. I promise it will help you a lot in all the other tasks in keeping the code format consistent.

File: src/main.js (hunk 10 above)
Comment (majac91) [reply src/main.js]: It does! The learn js course has basically nothing on scope so I very vaguely understand it, I'll have to look up another source

File: src/index.html (hunk 12 above)
Comment (majac91) [reply src/index.html]: Got it. If that's all, do you think we can move on with the next part of this project? I'm all out of ideas 😄

File: src/index.html [hunk 14]
```diff
@@ -3,14 +3,29 @@
...
+        <h2 class='navbar__subtitle'>a travel journal</h2>
+    </nav>
+
+
+    <main class="gallery container">
+        <div class="buttons container">
+            <h1 class=buttons__view>View:</h1>
+            <button class='button button__visited' type='button'>Visited</button>
+            <button class='button button__buckelist' type='button'>Bucketlist</button>
+            <button class='button button__all' type='button'>Show all</button>
```
Comment (sandrina-p) [inline src/index.html]: "View:" does not carry any semantics about heading, so it just ben just a simple span.

One tip about Accessibility. Buttons that serve as a "toggle" (have an active and not active state) should have an `aria-pressed="true"` attribute (true or false).  That way people using a Screen Reader can understand what's the active state. Here's an [article about aria-pressed](https://www.accessibility-developer-guide.com/examples/sensible-aria-usage/pressed/)

Hint: To add/remove an attribute with JS you use [`setAttribute`](https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute)

File: src/main.js [hunk 15]
```diff
@@ -5,170 +5,170 @@ var dateFormating = moment(myDate).format('LL');
 
 console.log(dateFormating);
 
-let destinations = [
+let destinationsList = [
```
Comment (sandrina-p) [inline src/main.js]: This file is getting huge. It's a very common practice to split code into different files, like "modules". My recommendation: Create a `scripts/` folder and move this `destinationList` to it.

At the end of the file you must write `export default destinationList` so that you can import it from other files


```js
/src/scripts/destinationList.js

const destinationsList = [ .... ]

export default destinationsList
```

```js
// src/main.js
import moment from "moment";
import destinationList from './scripts/destinationList.js';

...

```

**Review by sandrina-p — COMMENTED** [review]
It seems you learned a ton of new stuff with this project and this is just the beginning! 🎉 

I left some final feedback for you: After it you can merge this PR and create a new branch (from master) so that we can focus on the next step of this project: Create the page to add an item to the bucket list!

File: src/main.js [hunk 16]
```diff
@@ -177,70 +177,212 @@ let destinations = [
...
+      if(destination.visited === true) {
+        img.classList.add('visited');
+        figcaption.classList.add('visited');
+
+      }
+      else {
+        img.classList.add('bucketlist');
+        figcaption.classList.add('bucketlist');
+
+      }
```
Comment (sandrina-p) [inline src/main.js]: To simplify you could add `bucketlist` to `figcaption` instead. It would simplify the code and the CSS.

File: src/index.html (hunk 12 above)
Comment (sandrina-p) [reply src/index.html]: Yes, please check my latest comment (was posted a few minutes later) saying exactly that :D

File: src/main.js (hunk 13 above)
Comment (majac91) [reply src/main.js]: Just noticed one of them was disabled for whatever reason, should be fine now

File: src/index.html (hunk 14 above)
Comment (majac91) [reply src/index.html]:
```
    if (activeBtn) {
      activeBtn.classList.remove("active-button");
      current.setAttribute("aria-pressed", false);
    }
```
Why does this work to remove ```.active-button``` but not to set ```aria-pressed='false'```?

File: src/main.js (hunk 15 above)
Comment (majac91) [reply src/main.js]: This is the error I'm getting
Screenshot 2020-09-12 at 14 39 25

File: src/main.js (hunk 16 above)
Comment (majac91) [reply src/main.js]: Not sure what you mean, isn't this adding ```bucketlist``` to ```figcaption```?

``` figcaption.classList.add('bucketlist'); ```

Comment (sandrina-p) [reply src/main.js]: Sorry my bad, I meant to add it the parent, `figure`

File: src/index.html (hunk 14 above)
Comment (sandrina-p) [reply src/index.html]: It must be `activeBtn`, not `current` :)

Comment (majac91) [reply src/index.html]: omg I swear I tried that and it didn't work 🤣

Comment (majac91) [reply src/index.html]: closing this pr 🎉

File: package.json [hunk 17]
```diff
@@ -19,7 +19,8 @@
   },
   "homepage": "https://github.com/majac91/Travel_journal#readme",
   "dependencies": {
-    "moment": "^2.27.0"
+    "moment": "^2.27.0",
+    "src": "^1.1.2"
```
Comment (sandrina-p) [inline package.json]: Why did you need to install `src`?

Comment (majac91) [reply package.json]: Moment was in the npm tutorial you’ve sent me I think, I followed along

Comment (majac91) [reply package.json:23]: Or maybe  it was in the documentation for moment, I’m not home to check rn

Comment (majac91) [reply package.json]: Just checked the video  and it's not a part of the tutorial but it's also not something I remember installing separately. Didn't it come with other npm modules and should I do something about it?

Comment (sandrina-p) [reply package.json]: I guess you did `npm install src` without realizing it. It's the only explanation. You can uninstall that dependency with `npm uninstall src` :)
